### PR TITLE
Clang format and tidy shouldnt change file ownership when run locally.

### PR DESCRIPTION
**Pull request by jtattermusch**
Currently, the clang format and clang tidy scripts, when run locally on a checked-out copy of grpc, 
any updated files with be suddenly owned by root. This is because the docker container runs as root by default and it recreates the files being formatted - which makes them owned by root.
This can be mitigated by running docker as "current user" when invoking the local version of clang format script.

The changes shouldn't affect the clang-format in sanity suite.

@dgquintas, I assume folks on C-core have been seeing this problem too?

### CODE REVIEW

**Comment by jtattermusch** [comment]
CC @matt-kwong

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_fullstack_streaming_pump.BM_PumpStreamServerToClient_SockPair__8.counters.old: 1


[microbenchmarks] No significant performance differences
```

File: tools/distrib/clang_tidy_code.sh [hunk 1]
```diff
@@ -27,7 +27,9 @@ then
   docker build -t grpc_clang_tidy tools/dockerfile/grpc_clang_tidy
 
   # run clang-tidy against the checked out codebase
-  docker run -e TEST=$TEST -e CHANGED_FILES="$CHANGED_FILES" -e CLANG_TIDY_ROOT="/local-code" --rm=true -v "${REPO_ROOT}":/local-code -t grpc_clang_tidy /clang_tidy_all_the_things.sh "$@"
+  # when modifying the checked-out files, the current user will be impersonated
+  # so that the updated files don't end up being owned by "root".
+  docker run -e TEST=$TEST -e CHANGED_FILES="$CHANGED_FILES" -e CLANG_TIDY_ROOT="/local-code" --rm=true -v "${REPO_ROOT}":/local-code --user "$(id -u):$(id -g)" -t grpc_clang_tidy /clang_tidy_all_the_things.sh "$@"
```
Comment (mehrdada) [inline tools/distrib/clang_tidy_code.sh]: s/$TEST/"$TEST"

File: tools/distrib/clang_format_code.sh [hunk 2]
```diff
@@ -25,7 +25,9 @@ then
   docker build -t grpc_clang_format tools/dockerfile/grpc_clang_format
 
   # run clang-format against the checked out codebase
-  docker run -e TEST=$TEST -e CHANGED_FILES="$CHANGED_FILES" -e CLANG_FORMAT_ROOT="/local-code" --rm=true -v "${REPO_ROOT}":/local-code -t grpc_clang_format /clang_format_all_the_things.sh
+  # when modifying the checked-out files, the current user will be impersonated
+  # so that the updated files don't end up being owned by "root".
+  docker run -e TEST=$TEST -e CHANGED_FILES="$CHANGED_FILES" -e CLANG_FORMAT_ROOT="/local-code" --rm=true -v "${REPO_ROOT}":/local-code --user "$(id -u):$(id -g)" -t grpc_clang_format /clang_format_all_the_things.sh
```
Comment (mehrdada) [inline tools/distrib/clang_format_code.sh]: s/$TEST/"$TEST"

Comment (jtattermusch) [reply tools/distrib/clang_format_code.sh]: Done.

File: tools/distrib/clang_tidy_code.sh (hunk 1 above)
Comment (jtattermusch) [reply tools/distrib/clang_tidy_code.sh]: Done.

**Comment by jtattermusch** [comment]
Comments addressed, PTAL.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by dgquintas** [comment]
wow I never noticed this, or heard of anyone noticing this... in any case, thanks!

**Comment by vjpai** [comment]
I have seen this issue. I might need to improve my docker setup. I end up chown'ing stuff afterward.

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_fullstack_streaming_pump.BM_PumpStreamServerToClient_SockPair__512.opt.new: 1
    bm_fullstack_streaming_pump.BM_PumpStreamServerToClient_SockPair__64.opt.new: 1


[microbenchmarks] No significant performance differences
```

**Pull request merged**